query_alter_table.go

```diff
@@ -0,0 +1,409 @@
+package bun
+
+import "github.com/uptrace/bun/schema"
+
+type ifExists bool
+
+// AppendQuery adds IF EXISTS clause to the query, if its value is true. It always returns a nil error.
+func (ifexists ifExists) AppendQuery(_ schema.Formatter, b []byte) ([]byte, error) {
+	if !ifexists {
+		return b, nil
+	}
+	return append(b, "IF EXISTS "...), nil
+}
+
+type ifNotExists bool
+
+// AppendQuery adds IF NOT EXISTS clause to the query, if its value is true. It always returns a nil error.
+func (ifnotexists ifNotExists) AppendQuery(_ schema.Formatter, b []byte) ([]byte, error) {
+	if !ifnotexists {
+		return b, nil
+	}
+	return append(b, "IF NOT EXISTS "...), nil
+}
+
+type SubqueryAppender interface {
+	schema.QueryAppender
+	AppendSubquery(schema.Formatter, []byte) ([]byte, error)
+}
+
+type ChainableSubquery interface {
+	SubqueryAppender
+	chain()
+}
```

Comment:
Looks like you've come with some new design/idea for Bun query builder. It *could be* a nice idea, but let's describe it first and come to understanding how we will use it with other queries like SELECT/UPDATE/INSERT.

Reply:
...but I'd rather not design something new just for `ALTER TABLE` and instead try to solve the task with existing APIs.

Reply:
Agree.  
I've added some more thoughts about that [below](https://github.com/uptrace/bun/pull/726#issuecomment-1344649186).  

Going to refactor `ChainableSubquery` away in favor of the existing API :ok_hand:.  
It can stay in Git history, or I could describe it in a proposal, if you think it could be used in the rest of the lib.